roles/mysql/tasks/main.yml

```diff
@@ -51,9 +51,15 @@
   register: mysql_temp_password
   ignore_errors: yes
 
+- name: Update mysql root password with temp password
+  shell: 'mysql -e "SET PASSWORD = PASSWORD(''{{ mysql_root_password }}''); FLUSH PRIVILEGES;" --connect-expired-password -u root -i -p''{{ mysql_temp_password.stdout }}'''
+  when: mysql_temp_password.stdout != ''
+  register: mysqlPasswordUpdated
+  ignore_errors: yes
+
 - name: Update mysql root password
-  shell: 'mysql -e "SET PASSWORD = PASSWORD(''{{ mysql_root_password }}'');" --connect-expired-password -uroot -ip"{{ mysql_temp_password.stdout }}"'
-  ignore_errors: yes 
+  shell: 'mysql -e "UPDATE mysql.user SET authentication_string=PASSWORD(''{{mysql_root_password}}'') WHERE User=''root''; FLUSH PRIVILEGES;" -u root -i -p''{{mysql_old_root_password}}'''
```

Comment:
What will happen in below scenario
Lets say task 1 is Update mysql root password with temp password fails
and task2 is  Update mysql root password

when task1 fails mysql password is still temp one
and then second task will also fail as we couldn't connect using old password

Reply:
It will fail at task2 because task1 has ignore_errors yes. Password will not be updated.
I have not tested this scenario.